docs/oauth.rst

```diff
@@ -160,3 +160,49 @@ Here is a sample code for your customized authentication flow
 
 .. _`GetAuthUrl()`: /PyDrive2/pydrive2/#pydrive2.auth.GoogleAuth.GetAuthUrl
 .. _`Auth(code)`: /PyDrive2/pydrive2/#pydrive2.auth.GoogleAuth.Auth
+
+
+Authentication with a service account
+--------------------------------------
+
+A *service account* is a special type of Google account intended to represent a non-human user that needs to authenticate and be authorized to access data in Google APIs.
+
+Typically, service accounts are used in scenarios such as:
+
+- Running workloads on virtual machines (VMs).
+- Running workloads on on-premises workstations or data centers that call Google APIs.
+- Running workloads which are not tied to the lifecycle of a human user.
+
+If we use OAuth client ID we need to do one manual login into the account with `LocalWebserverAuth()`_
+if we use a service account the login is automatic.
+
+
+.. code-block:: python
+
+    from oauth2client.service_account import ServiceAccountCredentials
+    from pydrive2.auth import GoogleAuth
+    from pydrive2.drive import GoogleDrive
+
+    # credentials filename
+    CREDENTIALS_NAME = 'credentials.json'
+    SCOPES = ["https://www.googleapis.com/auth/drive"]
+
+
+    def login_with_service_account():
+        """
+        Google Drive service with a service account.
+        note: for the service account to work, you need to share the folder or files
+        with the service account email.
+
+        :return: drive service
+        """
+        # create an instance of GoogleAuth
+        gauth = GoogleAuth()
+        # tell it is a service account
+        gauth.auth_method = 'service'
```

Comment:
it can be simpler, https://github.com/iterative/PyDrive2/blob/main/pydrive2/fs/spec.py#L137-L147 

no need to set auth_method and credentials directly, I would say that we need to discourage even ppl from doing this

Reply:
Hey, sure, that's the minimum that I have in mind:

```python
settings = {
            "client_config_backend": "service",
            "service_config": {
                "client_json_file_path": "service-credentials.json",
            },
        }
auth = GoogleAuth(settings=settings)
```

alternatively, `client_json` could be used (instead for `client_json_file_path`) to feed the in-memory dict + some additional options: scope, etc, etc - this is similar to the regular workflow.

Does it help / enough to try and simplify? please, let me know.

Reply:
Your piece of code + `ServiceAuth()` method works well for me.
```python
settings = {
            "client_config_backend": "service",
            "service_config": {
                "client_json_file_path": "service-credentials.json",
            },
        }
auth = GoogleAuth(settings=settings)
auth.ServiceAuth()
```
I think its a very good and simple example. In a second example, I tried to include the `client-secrets.json` and use more functionality and works well.
```python
    settings = {
        "client_config_backend": "service",
        "service_config": {
            "client_json_dict": {
                "type": "service_account",
                "project_id": "sfe-tp",
                "private_key_id": "c941d7aXcXeddX1bf2Xbd80X1eXb1aXX3X5XbXX5",
                "private_key": "-----BEGIN PRIVATE KEY-----XXXX-----END PRIVATE KEY-----",
                "client_email": "xxxx-upload@xxxx.gserviceaccount.com",
                "client_id": "1049xx94805xxxx59xxxx306",
                "auth_uri": "https://accounts.google.com/o/oauth2/auth",
                "token_uri": "https://oauth2.googleapis.com/token",
                "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
                "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/sxxxx.com"
            },
        },
        "save_credentials": True,
        "save_credentials_backend": "file",
        "save_credentials_file": "credentials.json",
        "get_refresh_token": True,
        "oauth_scope": ["https://www.googleapis.com/auth/drive"]
    }
    gauth = GoogleAuth(settings=settings)
    gauth.ServiceAuth()
```
What example do you think its better? Also i couldnt test `client_json` because expects a string and idk how to transform the `client-secrets.json` to a one string.